lib/puppet_x/vmware/mapper/virtual_disk_map.rb

```diff
@@ -0,0 +1,65 @@
+# Copyright 2015 VMware, Inc.
+
+require 'set'
+
+module PuppetX::VMware::Mapper
+
+  class VirtualDiskMap < Map
+    def initialize
+      @initTree = {
+        Node => NodeData[
+          :node_type => 'VirtualDisk',
+        ],
+        :key           => LeafData[
+          :desc          => 'Only used internally when a update or removal to an existing monitor',
+        ],
+        :controllerKey => LeafData[
+          :desc          => 'Object key for the controller object for this device. This property contains the key property value of the controller device object.',
+          :prop_name     => 'controller'
+        ],
+        :capacityInKB => LeafData[
```

Comment:
It doesn't work. I originally went down the road of using capacityInKB and I got back failures. After a couple days in the weeds, I went back to capacityInBytes. I think it might be a RbVmomi issue but I am not 100% sure right now.